source/types.ts

```diff
@@ -25,43 +25,39 @@ export interface CreateComponentOptions {
   onAborted?: () => unknown;
 }
 
-export interface Entity {
-  id: string;
-  __entity_type__: string;
-}
 interface ResponseMetadata {
   next: {
     offset: number | null;
   };
 }
 export interface SearchOptions {
   expression: string;
-  entityType: string;
+  entityType: EntityType;
   terms?: string[];
   contextId?: string;
   objectTypeIds?: string[];
 }
 
-export interface QueryResponse<T = Data> {
-  data: T[];
+export interface QueryResponse<K extends EntityType = EntityType> {
+  data: EntityData<K>[]; // Later this needs to take multiple different K and give the correct types
```

Comment:
can you expand on what this comment means and how it should be resolved?

Reply:
I didn't add this actually. I think @gismya could chime in?

Reply:
A call can contain multiple query operations with different entity types.. But writing this I realize that would be handled in the tuple of the call and not here. I'll remove the comment.